### PR TITLE
Add 1upkeyboards Super16v3

### PR DESCRIPTION
**Pull request by ziptyze**
Addition of the 1upkeyboards Super16v3 macropad

## Description

The Super16v3 is the newest version of 1upkeyboards 4x4 macropad. It features an rp2040 controller, direct i/o for all switches, the option to have up to four encoders in the top row, and in-switch and underglow addressable RGB leds.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/1upkeyboards/1upsuper16v3/readme.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# 1upocarina
+
+This keyboard is the Super16v3 from 1upkeyboards. It is a 4x4 macropad with the option for up to four encoders in the top row. The keypad features in-switch per-key addressable RGB leds, as well as 32 1mm underglow RGB leds for very bright and smooth lighting effects. The Super16v3 uses an rp2040 microcontroller and direct pins instead of a matrix.
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1upsuper16v3/readme.md:3]:
```suggestion
# Super16v3

This keyboard is the Super16v3 from 1upkeyboards, a 4x4 macropad with the option for up to four encoders in the top row.  Notable features include in-switch per-key addressable RGB LEDs and thirty two 1mm underglow RGB LEDs; for bright and smooth lighting effects. The Super16v3 utilises an rp2040 microcontroller, wired using direct pins.
```

File: keyboards/1upkeyboards/1upsuper16v3/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    RGB_MOD,   RGB_HUD,    RGB_SAD,    RGB_VAD,
+    RGB_SPD,   RGB_SPI,    KC_TRNS,    KC_TRNS,
+    KC_TRNS,   KC_TRNS,    KC_TRNS,    QK_BOOT
+  ),
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
+    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS)  },
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1upsuper16v3/keymaps/via/keymap.c:38]: remove, possibly problematic, commas
```suggestion
  )
};

#if defined(ENCODER_MAP_ENABLE)
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS)  }
```

File: keyboards/1upkeyboards/1upsuper16v3/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    RGB_MOD,   RGB_HUD,    RGB_SAD,    RGB_VAD,
+    RGB_SPD,   RGB_SPI,    KC_TRNS,    KC_TRNS,
+    KC_TRNS,   KC_TRNS,    KC_TRNS,    QK_BOOT
+  ),
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
+    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS)  },
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1upsuper16v3/keymaps/default/keymap.c:38]: remove, possibly problematic, commas
```suggestion
  )
};

#if defined(ENCODER_MAP_ENABLE)
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS)  }
```

File: keyboards/1upkeyboards/1upsuper16v3/info.json [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        { "flags": 4, "matrix": [0, 2], "x": 140, "y": 8 },
+
+        { "flags": 2, "x": 220, "y": 4 },
+        { "flags": 2, "x": 210, "y": 1 },
+        { "flags": 2, "x": 182, "y": 1 },
+        { "flags": 2, "x": 154, "y": 1 },
+        { "flags": 2, "x": 126, "y": 1 }
+
+        ]
+    },
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1upsuper16v3/info.json:172]: matching community layout(s) identified
```suggestion
    },
    "community_layouts": [ "ortho_4x4" ],
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Comment by dunk2k** [comment]
@ziptyze 
could directory `1upsuper16v3` be renamed to `super16v3`?

File: keyboards/1upkeyboards/1upsuper16v3/info.json [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+                { "label": "12", "matrix": [0, 6], "w": 1, "x": 2, "y": 1 },
+                { "label": "13", "matrix": [0, 7], "w": 1, "x": 3, "y": 1 },
+                { "label": "20", "matrix": [0, 8], "w": 1, "x": 0, "y": 2 },
+                { "label": "21", "matrix": [0, 9], "w": 1, "x": 1, "y": 2 },
+                { "label": "22", "matrix": [0, 10], "w": 1, "x": 2, "y": 2 },
+                { "label": "23", "matrix": [0, 11], "w": 1, "x": 3, "y": 2 },
+                { "label": "30", "matrix": [0, 12], "w": 1, "x": 0, "y": 3 },
+                { "label": "31", "matrix": [0, 13], "w": 1, "x": 1, "y": 3 },
+                { "label": "32", "matrix": [0, 14], "w": 1, "x": 2, "y": 3 },
+                { "label": "33", "matrix": [0, 15], "w": 1, "x": 3, "y": 3 }
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1upsuper16v3/info.json:191]: `w: 1` is default key width
```suggestion
                { "label": "00", "matrix": [0, 0], "x": 0, "y": 0 },
                { "label": "01", "matrix": [0, 1], "x": 1, "y": 0 },
                { "label": "02", "matrix": [0, 2], "x": 2, "y": 0 },
                { "label": "03", "matrix": [0, 3], "x": 3, "y": 0 },
                { "label": "10", "matrix": [0, 4], "x": 0, "y": 1 },
                { "label": "11", "matrix": [0, 5], "x": 1, "y": 1 },
                { "label": "12", "matrix": [0, 6], "x": 2, "y": 1 },
                { "label": "13", "matrix": [0, 7], "x": 3, "y": 1 },
                { "label": "20", "matrix": [0, 8], "x": 0, "y": 2 },
                { "label": "21", "matrix": [0, 9], "x": 1, "y": 2 },
                { "label": "22", "matrix": [0, 10], "x": 2, "y": 2 },
                { "label": "23", "matrix": [0, 11], "x": 3, "y": 2 },
                { "label": "30", "matrix": [0, 12], "x": 0, "y": 3 },
                { "label": "31", "matrix": [0, 13], "x": 1, "y": 3 },
                { "label": "32", "matrix": [0, 14], "x": 2, "y": 3 },
                { "label": "33", "matrix": [0, 15], "x": 3, "y": 3 }
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1upsuper16v3/info.json [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+                { "label": "12", "matrix": [0, 6], "x": 2, "y": 1 },
+                { "label": "13", "matrix": [0, 7], "x": 3, "y": 1 },
+                { "label": "20", "matrix": [0, 8], "x": 0, "y": 2 },
+                { "label": "21", "matrix": [0, 9], "x": 1, "y": 2 },
+                { "label": "22", "matrix": [0, 10], "x": 2, "y": 2 },
+                { "label": "23", "matrix": [0, 11], "x": 3, "y": 2 },
+                { "label": "30", "matrix": [0, 12], "x": 0, "y": 3 },
+                { "label": "31", "matrix": [0, 13], "x": 1, "y": 3 },
+                { "label": "32", "matrix": [0, 14], "x": 2, "y": 3 },
+                { "label": "33", "matrix": [0, 15], "x": 3, "y": 3 }
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1upsuper16v3/info.json:192]: entirely optional - puts the direct pin matrix into same shape as physical layout
```suggestion
            ["GP23", "GP24", "GP4", "GP5"],
            ["GP19", "GP27", "GP1", "GP8"],
            ["GP18", "GP28", "GP0", "GP9"],
            ["GP17", "GP16", "GP15", "GP11" ]
        ]
    },
    "ws2812": {
        "pin": "GP29",
        "driver": "vendor"
    },
    "rgb_matrix": {
        "animations": {
            "gradient_up_down": true,
            "gradient_left_right": true,
            "breathing": true,
            "band_sat": true,
            "band_val": true,
            "band_pinwheel_sat": true,
            "band_pinwheel_val": true,
            "band_spiral_sat": true,
            "band_spiral_val": true,
            "cycle_all": true,
            "cycle_left_right": true,
            "cycle_up_down": true,
            "rainbow_moving_chevron":true,
            "cycle_out_in": true,
            "cycle_out_in_dual": true,
            "cycle_pinwheel": true,
            "cycle_spiral": true,
            "dual_beacon": true,
            "rainbow_beacon": true,
            "rainbow_pinwheels": true,
            "raindrops": true,
            "jellybean_raindrops": true,
	        "hue_breathing": true,
            "hue_pendulum": true,
            "hue_wave": true,
            "pixel_rain": true,
            "pixel_flow": true,
            "typing_heatmap": true,
            "digital_rain": true,
            "solid_reactive_simple": true,
	        "solid_reactive": true,
            "solid_reactive_wide": true,
            "solid_reactive_multiwide": true,
            "solid_reactive_cross": true,
            "solid_reactive_multicross": true,
            "solid_reactive_nexus": true,
            "solid_reactive_multinexus": true,
            "splash": true,
            "multisplash": true,
            "solid_splash": true,
            "solid_multisplash": true
        },
        "driver": "WS2812",
        "layout": [

        { "flags": 2, "x": 99, "y": 1 },
        { "flags": 2, "x": 70, "y": 1 },
        { "flags": 2, "x": 42, "y": 1 },
        { "flags": 2, "x": 14, "y": 1 },
        { "flags": 2, "x": 4, "y": 4 },
        { "flags": 2, "x": 4, "y": 12 },

        { "flags": 4, "matrix": [0, 0], "x": 28, "y": 8 },
        { "flags": 4, "matrix": [0, 1], "x": 84, "y": 8 },

        { "flags": 2, "x": 4, "y": 20 },
        { "flags": 2, "x": 4, "y": 28 },

        { "flags": 4, "matrix": [0, 4], "x": 28, "y": 24 },
        { "flags": 4, "matrix": [0, 5], "x": 84, "y": 24 },

        { "flags": 2, "x": 4, "y": 36 },
        { "flags": 2, "x": 4, "y": 44 },

        { "flags": 4, "matrix": [0, 8], "x": 28, "y": 40 },
        { "flags": 4, "matrix": [0, 9], "x": 84, "y": 40 },

        { "flags": 2, "x": 4, "y": 52 },
        { "flags": 2, "x": 4, "y": 60 },
        { "flags": 2, "x": 14, "y": 63 },

        { "flags": 4, "matrix": [0, 12], "x": 28, "y": 56 },

        { "flags": 2, "x": 42, "y": 63 },
        { "flags": 2, "x": 70, "y": 63 },

        { "flags": 4, "matrix": [0, 13], "x": 84, "y": 56 },

        { "flags": 2, "x": 99, "y": 63 },
        { "flags": 2, "x": 126, "y": 63 },

        { "flags": 4, "matrix": [0, 14], "x": 140, "y": 56 },

        { "flags": 2, "x": 154, "y": 63 },
        { "flags": 2, "x": 182, "y": 63 },

        { "flags": 4, "matrix": [0, 15], "x": 196, "y": 56 },

        { "flags": 2, "x": 210, "y": 63 },
        { "flags": 2, "x": 220, "y": 60 },
        { "flags": 2, "x": 220, "y": 52 },
        { "flags": 2, "x": 220, "y": 44 },

        { "flags": 4, "matrix": [0, 11], "x": 196, "y": 40 },
        { "flags": 4, "matrix": [0, 10], "x": 140, "y": 40 },

        { "flags": 2, "x": 220, "y": 36 },
        { "flags": 2, "x": 220, "y": 28 },

        { "flags": 4, "matrix": [0, 7], "x": 196, "y": 24 },
        { "flags": 4, "matrix": [0, 6], "x": 140, "y": 24 },

        { "flags": 2, "x": 220, "y": 20 },
        { "flags": 2, "x": 220, "y": 12 },

        { "flags": 4, "matrix": [0, 3], "x": 196, "y": 8 },
        { "flags": 4, "matrix": [0, 2], "x": 140, "y": 8 },

        { "flags": 2, "x": 220, "y": 4 },
        { "flags": 2, "x": 210, "y": 1 },
        { "flags": 2, "x": 182, "y": 1 },
        { "flags": 2, "x": 154, "y": 1 },
        { "flags": 2, "x": 126, "y": 1 }

        ]
    },
    "community_layouts": [ "ortho_4x4" ],
    "layouts": {
        "LAYOUT_ortho_4x4": {
            "layout": [
                { "label": "00", "matrix": [0, 0], "x": 0, "y": 0 },
                { "label": "01", "matrix": [0, 1], "x": 1, "y": 0 },
                { "label": "02", "matrix": [0, 2], "x": 2, "y": 0 },
                { "label": "03", "matrix": [0, 3], "x": 3, "y": 0 },
                { "label": "10", "matrix": [1, 0], "x": 0, "y": 1 },
                { "label": "11", "matrix": [1, 1], "x": 1, "y": 1 },
                { "label": "12", "matrix": [1, 2], "x": 2, "y": 1 },
                { "label": "13", "matrix": [1, 3], "x": 3, "y": 1 },
                { "label": "20", "matrix": [2, 0], "x": 0, "y": 2 },
                { "label": "21", "matrix": [2, 1], "x": 1, "y": 2 },
                { "label": "22", "matrix": [2, 2], "x": 2, "y": 2 },
                { "label": "23", "matrix": [2, 3], "x": 3, "y": 2 },
                { "label": "30", "matrix": [3, 0], "x": 0, "y": 3 },
                { "label": "31", "matrix": [3, 1], "x": 1, "y": 3 },
                { "label": "32", "matrix": [3, 2], "x": 2, "y": 3 },
                { "label": "33", "matrix": [3, 3], "x": 3, "y": 3 }
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1upsuper16v3/info.json [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+
+        { "flags": 4, "matrix": [0, 3], "x": 196, "y": 8 },
+        { "flags": 4, "matrix": [0, 2], "x": 140, "y": 8 },
+
+        { "flags": 2, "x": 220, "y": 4 },
+        { "flags": 2, "x": 210, "y": 1 },
+        { "flags": 2, "x": 182, "y": 1 },
+        { "flags": 2, "x": 154, "y": 1 },
+        { "flags": 2, "x": 126, "y": 1 }
+
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1upsuper16v3/info.json:173]: Could you format this section please?

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1upsuper16v3/info.json [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+        { "flags": 2, "x": 220, "y": 28 },
+
+        { "flags": 4, "matrix": [0, 7], "x": 196, "y": 24 },
+        { "flags": 4, "matrix": [0, 6], "x": 140, "y": 24 },
+
+        { "flags": 2, "x": 220, "y": 20 },
+        { "flags": 2, "x": 220, "y": 12 },
+
+        { "flags": 4, "matrix": [0, 3], "x": 196, "y": 8 },
+        { "flags": 4, "matrix": [0, 2], "x": 140, "y": 8 },
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1upsuper16v3/info.json:166]: fixes CI Build failures after matrix re-jig
```suggestion
        { "flags": 4, "matrix": [0, 0], "x": 28, "y": 8 },
        { "flags": 4, "matrix": [0, 1], "x": 84, "y": 8 },

        { "flags": 2, "x": 4, "y": 20 },
        { "flags": 2, "x": 4, "y": 28 },

        { "flags": 4, "matrix": [1, 0], "x": 28, "y": 24 },
        { "flags": 4, "matrix": [1, 1], "x": 84, "y": 24 },

        { "flags": 2, "x": 4, "y": 36 },
        { "flags": 2, "x": 4, "y": 44 },

        { "flags": 4, "matrix": [2, 0], "x": 28, "y": 40 },
        { "flags": 4, "matrix": [2, 1], "x": 84, "y": 40 },

        { "flags": 2, "x": 4, "y": 52 },
        { "flags": 2, "x": 4, "y": 60 },
        { "flags": 2, "x": 14, "y": 63 },

        { "flags": 4, "matrix": [3, 0], "x": 28, "y": 56 },

        { "flags": 2, "x": 42, "y": 63 },
        { "flags": 2, "x": 70, "y": 63 },

        { "flags": 4, "matrix": [3, 1], "x": 84, "y": 56 },

        { "flags": 2, "x": 99, "y": 63 },
        { "flags": 2, "x": 126, "y": 63 },

        { "flags": 4, "matrix": [3, 2], "x": 140, "y": 56 },

        { "flags": 2, "x": 154, "y": 63 },
        { "flags": 2, "x": 182, "y": 63 },

        { "flags": 4, "matrix": [3, 3], "x": 196, "y": 56 },

        { "flags": 2, "x": 210, "y": 63 },
        { "flags": 2, "x": 220, "y": 60 },
        { "flags": 2, "x": 220, "y": 52 },
        { "flags": 2, "x": 220, "y": 44 },

        { "flags": 4, "matrix": [2, 3], "x": 196, "y": 40 },
        { "flags": 4, "matrix": [2, 2], "x": 140, "y": 40 },

        { "flags": 2, "x": 220, "y": 36 },
        { "flags": 2, "x": 220, "y": 28 },

        { "flags": 4, "matrix": [1, 3], "x": 196, "y": 24 },
        { "flags": 4, "matrix": [1, 2], "x": 140, "y": 24 },

        { "flags": 2, "x": 220, "y": 20 },
        { "flags": 2, "x": 220, "y": 12 },

        { "flags": 4, "matrix": [0, 3], "x": 196, "y": 8 },
        { "flags": 4, "matrix": [0, 2], "x": 140, "y": 8 },
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/1upkeyboards/1upsuper16v3/info.json (hunk 7 above)
Comment (ziptyze) [reply keyboards/1upkeyboards/1upsuper16v3/info.json]: In what way? I put the spaces in as it is to make it easier to visualize the physical path of the LEDs. It jumps back and forth between in-switch and underglow.

**Comment by ziptyze** [comment]
> @ziptyze could directory `1upsuper16v3` be renamed to `super16v3`?

We had been actively trying to move toward having the name of the product and directories start with 1up, so anything we're adding is following that naming convention, and at some point when I get around to it we want to do that to the rest.

File: keyboards/1upkeyboards/1upsuper16v3/info.json (hunk 7 above)
Comment (ziptyze) [reply keyboards/1upkeyboards/1upsuper16v3/info.json]: Ok, I made some changes. Let me know if it is what you're looking for.

File: keyboards/1upkeyboards/1upsuper16v3/keymaps/default/keymap.c [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    RGB_MOD,   RGB_HUD,    RGB_SAD,    RGB_VAD,
+    RGB_SPD,   RGB_SPI,    KC_TRNS,    KC_TRNS,
+    KC_TRNS,   KC_TRNS,    KC_TRNS,    QK_BOOT
+  )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
+    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS)  }
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1upsuper16v3/keymaps/default/keymap.c:38]: for the love of formatting
```suggestion
    [0] = LAYOUT_ortho_4x4(
        KC_P7, KC_P8, KC_P9,   KC_PMNS,
        KC_P4, KC_P5, KC_P6,   KC_PPLS,
        KC_P1, KC_P2, KC_P3,   KC_PSLS,
        MO(1), KC_P0, KC_PDOT, KC_PENT
    ),
    [1] = LAYOUT_ortho_4x4(
        RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI,
        RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD,
        RGB_SPD, RGB_SPI, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, QK_BOOT
    )
};

#if defined(ENCODER_MAP_ENABLE)
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
    [1] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) }
```

File: keyboards/1upkeyboards/1upsuper16v3/keymaps/via/keymap.c [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+    RGB_MOD,   RGB_HUD,    RGB_SAD,    RGB_VAD,
+    RGB_SPD,   RGB_SPI,    KC_TRNS,    KC_TRNS,
+    KC_TRNS,   KC_TRNS,    KC_TRNS,    QK_BOOT
+  )
+};
+
+#if defined(ENCODER_MAP_ENABLE)
+const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
+    [0] =  { ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU),      ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
+    [1] =  { ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS),      ENCODER_CCW_CW(KC_TRNS, KC_TRNS)  }
```
Comment (dunk2k) [inline keyboards/1upkeyboards/1upsuper16v3/keymaps/via/keymap.c:38]: for the love of formatting
```suggestion
    [0] = LAYOUT_ortho_4x4(
        KC_P7, KC_P8, KC_P9,   KC_PMNS,
        KC_P4, KC_P5, KC_P6,   KC_PPLS,
        KC_P1, KC_P2, KC_P3,   KC_PSLS,
        MO(1), KC_P0, KC_PDOT, KC_PENT
    ),
    [1] = LAYOUT_ortho_4x4(
        RGB_TOG, RGB_HUI, RGB_SAI, RGB_VAI,
        RGB_MOD, RGB_HUD, RGB_SAD, RGB_VAD,
        RGB_SPD, RGB_SPI, KC_TRNS, KC_TRNS,
        KC_TRNS, KC_TRNS, KC_TRNS, QK_BOOT
    )
};

#if defined(ENCODER_MAP_ENABLE)
const uint16_t PROGMEM encoder_map[][NUM_ENCODERS][NUM_DIRECTIONS] = {
    [0] = { ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_VOLD, KC_VOLU), ENCODER_CCW_CW(KC_VOLD, KC_VOLU)  },
    [1] = { ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS), ENCODER_CCW_CW(KC_TRNS, KC_TRNS) }
```

**Review by dunk2k — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by dunk2k — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Review by dunk2k — APPROVED** [review]

File: keyboards/1upkeyboards/1upsuper16v3/readme.md
Comment (dunk2k) [inline keyboards/1upkeyboards/1upsuper16v3/readme.md:1]: Would an image (link) of this board/PCB be possible?

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**